corehq/apps/app_manager/models.py

```diff
@@ -5720,6 +5720,7 @@ def _hash(val):
 
         previous_version = self.get_previous_version()
         if previous_version:
+            force_new_version = self.build_profiles != previous_version.build_profiles
```

Comment:
@calellowitz can you validate this change?  Previously this was only being checked when `set_form_versions` was called through `create_all_files`, and now it'll always be checked.

Reply:
In what other instances is this called? I can't imagine this would break anything, and it seems unlikely to change behavior, although I guess there is a small possibility that it could increment version numbers unnecessarily which would cause the next update to download a form it did not have to.

Reply:
It was called during several of the download views, though turned out that most were noop-ing, so I removed them.  Now it's only called through `create_all_files` and in the `download_suite` view.  Since previous_version is being fetched just before, I think this'd be the correct behavior anyways (before this PR, `previous_version` was often not provided, so I could see also not checking for a version update).